File: AwsServer/src/main/java/com/awsServer/security/post/Post.java
package com.awsServer.security.post;

import com.awsServer.security.forum.Forum;
import com.awsServer.security.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "_post")
public class Post
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String title;
    private String body;
    private Date date;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "forum_id")
    private Forum forum;
}

File: AWSApp/Shared/Model/Post.cs
namespace Shared.Model;

public class Post
{
    public int Id { get; set; }
    public User Username { get; }
    public string Title { get; }

    public string Body { get; set; }

    public Post(User username, string title, string body)
    {
        Username = username;
        Title = title;
        Body = body;

    }
}